aws-predictions/src/androidTest/java/com/amplifyframework/predictions/aws/AWSPredictionsIdentifyEntitiesTest.java

```diff
@@ -0,0 +1,94 @@
+
+import android.content.Context;
+import android.graphics.Bitmap;
+
+import com.amplifyframework.predictions.PredictionsCategory;
+import com.amplifyframework.predictions.aws.test.R;
+import com.amplifyframework.predictions.models.EmotionType;
+import com.amplifyframework.predictions.models.EntityDetails;
+import com.amplifyframework.predictions.models.GenderBinaryType;
+import com.amplifyframework.predictions.models.IdentifyActionType;
+import com.amplifyframework.predictions.result.IdentifyEntitiesResult;
+import com.amplifyframework.testutils.Assets;
+import com.amplifyframework.testutils.FeatureAssert;
+import com.amplifyframework.testutils.sync.SynchronousMobileClient;
+import com.amplifyframework.testutils.sync.SynchronousPredictions;
+import com.amplifyframework.util.CollectionUtils;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+
+import static androidx.test.core.app.ApplicationProvider.getApplicationContext;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests that Predictions identify delivers a non-null
+ * entity detection result for valid input.
+ */
+public final class AWSPredictionsIdentifyEntitiesTest {
+
+    private static final IdentifyActionType TYPE = IdentifyActionType.DETECT_ENTITIES;
+
+    private SynchronousPredictions predictions;
+
+    /**
+     * Configure Predictions category before each test.
+     * @throws Exception if mobile client initialization fails
+     */
+    @Before
+    public void setUp() throws Exception {
+        Context context = getApplicationContext();
+
+        // Set up Auth
+        SynchronousMobileClient.instance().initialize();
+
+        // Delegate to Predictions category
+        PredictionsCategory asyncDelegate =
+                TestPredictionsCategory.create(context, R.raw.amplifyconfiguration);
+        predictions = SynchronousPredictions.delegatingTo(asyncDelegate);
+    }
+
+    /**
+     * Assert general entity detection works.
+     * @throws Exception if prediction fails
+     */
+    @Test
+    @SuppressWarnings("MagicNumber") // Jeff Bezos' current age
+    public void testIdentifyEntities() throws Exception {
+        final Bitmap image = Assets.readAsBitmap("jeff_bezos.jpg");
+
+        // Identify the entity inside given image and assert non-null result.
+        IdentifyEntitiesResult result = (IdentifyEntitiesResult) predictions.identify(TYPE, image);
+        assertNotNull(result);
+
+        // Assert that at least one entity is detected
+        assertFalse(CollectionUtils.isNullOrEmpty(result.getEntities()));
+        EntityDetails entity = result.getEntities().get(0);
+
+        // Assert features from detected entity
+        FeatureAssert.assertMatches(GenderBinaryType.MALE, entity.getGender());
+        FeatureAssert.assertMatches(EmotionType.HAPPY, Collections.max(entity.getEmotions()));
```

Comment:
Wasn't there a `getPredominantSentiment()` or something like that? Should we provide an API for this, `getPredominantEmotion()` or something?

The thing about test code is that it tends to look like what a customer will do. If the customer just wants to know "Is Jeff Bezos happy?" (Insert joke here...)

(... back from joke) what about like `FeatureAssert.assertMatches(EmotionType.HAPPY, entity.getPredominantEmotion())`? // Returns the one with most confidence?

Reply:
it would be convenient in a way... but that would unfortunately break the platform parity. 

To be completely fair, iOS's implementation of `interpret` provides BOTH predominant sentiment AND the map of all other sentiments (and their confidence scores). I'll leave it as for now, but it would certainly be a nice addition in the future on both platforms.